x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/add_metadata_filter_button.tsx

```diff
@@ -37,9 +37,28 @@ export const AddMetadataFilterButton = ({ item }: AddMetadataFilterButtonProps)
         query: { filterManager: filterManagerService },
       },
       notifications: { toasts: toastsService },
+      telemetry,
     },
   } = useKibanaContextForPlugin();
 
+  const reportHostFlyoutAddFilter = useCallback(
+    ({ fieldName }) => {
+      telemetry.reportHostFlyoutAddFilter({
+        field_name: fieldName,
+      });
+    },
+    [telemetry]
+  );
+
+  const reportHostFlyoutRemoveFilter = useCallback(
+    ({ fieldName }) => {
+      telemetry.reportHostFlyoutAddFilter({
+        field_name: fieldName,
+      });
+    },
+    [telemetry]
+  );
+
```

Comment:
**nit:** there's no need to wrap these two functions with `useCallback` since they won't be passed as a reference down in the tree and no component memoization is applied

Reply:
yeah, makes sense, removed the `useCallback` here.